src/libs/blueprint/blueprint_mesh.cpp

```diff
@@ -94,8 +133,504 @@ bool
+                res = false;
+            }
+        }
+    }
+    
+    // optional: "fields", each child must conform to "mesh::field"
+    if(n.has_path("fields"))
+    {
+        NodeConstIterator itr = n["fields"].children();
+    
+        while(itr.has_next())
+        {
+            const Node &chld = itr.next();
+            const std::string chld_name = itr.path();
+
+            if(!field::verify(chld,info["fields/" + chld_name]))
+            {
+                info["errors"].append().set(chld_name 
+                                    + " is not a valid blueprint::mesh::field");
+                res = false;
+            }
+
+            // make sure the field references a valid topology
+            std::string topo_name = chld["topology"].as_string();
+        
+            if(!n["topologies"].has_child(topo_name))
+            {
+                std::ostringstream oss;
+                oss << "field "
+                    << "\"" << chld_name  << "\" "
+                    << "references a non-existent topology "
+                    << "\"" << topo_name  << "\" ";
+                info["errors"].append().set(oss.str());
+                res = false;
+            }
+            else if(info["topologies"][topo_name]["valid"].as_string() != "true")
+            {
+                std::ostringstream oss;
+                oss << "field "
+                    << "\"" << chld_name  << "\" "
+                    << "references a invalid topology "
+                    << "\"" << topo_name << "\" ";
+                info["errors"].append().set(oss.str());
+                res = false;
+            }
+        }
+    }
+    
+    if(res)
+    {
+        info["valid"] = "true";
+    }
+    else
+    {
+        info["valid"] = "false";
+    }
+
+    return res;
+}
+
+//-----------------------------------------------------------------------------
+// blueprint::mesh::coordset protocol interface
+//-----------------------------------------------------------------------------
+namespace coordset
+{
+//-----------------------------------------------------------------------------
+bool
+verify(const Node &coordset)
+{
+    Node info;
+    return verify(coordset,info);
+}
+
+//-----------------------------------------------------------------------------
+bool
+verify_coordset_uniform(const Node &coordset,
+                        Node &info)
+{
+    bool res = true;
+
+    if(!coordset.has_child("dims"))
+    {
+        std::string msg = "missing uniform coordset child \"value\"";
+        info["errors"].append().set(msg);
+        res = false;
+    }
+    else // dims needs at least one child ("i") 
+    {
+        const Node &dims = coordset["dims"];
+        
+        if(!dims.has_child("i"))
+        {
+            std::string msg ="missing coordset child \"dims\"";
+            info["errors"].append().set(msg);
+            res = false;
+        }
+        else if(!dims["i"].dtype().is_number())
```

Comment:
sure, feel free to explore it more. I don't know if we will have too much time to get into the details tomorrow, but if you have a prototype of how it would look that would help frame the discussion.